### DISCUSSION TITLE
How to make opus transcoding work along with seeking in Android clients?

### DISCUSSION THREAD

**Discussion by vctls** [ampache/ampache#4083] [Q&A]
First off, the reason I'd like to use opus is because it has one of the lowest "transparent" bitrates.

Most of my music in flac. I've enabled transcoding in the server configuration with the following options:
```
transcode_flac = "allowed"
encode_target = "opus"
transcode_cmd = "ffmpeg"
encode_args_opus = "-vn -b:a %BITRATE% -c:a libopus -compression_level 10 -f ogg pipe:1"
```
This last line is the default. As is most of the rest of my configuration.

Enabling opus transcoding completely broke Power Ampache 2, so I switched to Tempo.
The transcodes mostly work in Tempo, but by default it disables the seek bar. Also, playback sometimes jumps back to the beginning of the track.
Tempo has an option to "Estimate content length". When enabled, the seek bar works again for most tracks, but some tracks just fail to play completely. This is what I have in the Ampache logs:
```
2024-11-28T21:33:32+00:00 [victor] (Ampache\Module\Playback\Stream) -> Transcode command prefix: exec 
2024-11-28T21:33:32+00:00 [victor] (Ampache\Module\Playback\Stream) -> Command: ffmpeg Arguments: -i %FILE% -vn -b:a %BITRATE% -c:a libopus -compression_level 10 -f ogg pipe:1
2024-11-28T21:33:32+00:00 [victor] (Ampache\Module\Application\Playback\PlayAction) -> Content-Range doesn't start from 0, stats should already be registered previously; not collecting stats
2024-11-28T21:33:35+00:00 [victor] (Ampache\Module\Playback\Stream) -> Process is not running, kill skipped.
2024-11-28T21:33:35+00:00 [victor] (Ampache\Module\Application\Playback\PlayAction) -> Stream ended at 4128000 (3681426) bytes out of 4128000
```
Is there something I can tweak in the transcoding options to make everything work?

**Comment by lachlan-00** [comment]
switch to play2 in streaming options `preferences.php?tab=streaming`

![image](https://github.com/user-attachments/assets/e5fb350e-1946-4169-925e-c21662eef92b)

i flip the streaming buffer around and that will probably fix you

**Reply by vctls** [reply]
@lachlan-00 thank you, it seems to be working better!
I still had some delays initiating playback. Will test some more and get back to you.

**Reply by vctls** [reply]
Playback is indeed working just fine, at least in Tempo.  
I suppose the longer initial delay (a few seconds) is to be expected with this buffering method. Since queued tracks buffer in advance, it's not an issue when playing lists. Seeking is a bit slow but it's working as well.  
For some reason Power Ampache 2 still can't play anything, maybe it just doesn't handle opus correctly. Ultrasonic doesn't work either. Oh well. I'll stick with Tempo.  

Thank you!

**Reply by vctls** [reply]
Huh, well actually I just had another issue. On track just refused to play.  
It seems to happen with all tracks exceeding a certain length.

I think I'm just going to give up opus and go back to good old mp3. :shrug:   
There are just too many issues with it.
